app/Http/Controllers/CampaignsController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Campaign;
+use Illuminate\Http\Request;
+use Rogue\Http\Transformers\CampaignTransformer;
+
+class CampaignsController extends ApiController
+{
+    /**
+     * @var Rogue\Http\Transformers\SignupTransformer;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     */
+    public function __construct()
+    {
+        $this->transformer = new CampaignTransformer;
+
+        // $this->middleware('auth');
+        // $this->middleware('role:admin,staff', ['only' => ['store', 'update', 'destroy']]);
+        // $this->middleware('scopes:write', ['only' => ['store', 'update', 'destroy']]);
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $query = $this->newQuery(Campaign::class);
+
+        return $this->paginatedCollection($query, $request);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'internal_title' => 'required|string',
+            'start_date' => 'required|date',
+            'end_date' => 'required|date',
+        ]);
+
+        // Check to see if the campaign exists before creating one.
+        $campaign = Campaign::where([
+            'internal_title' => $request['internal_title'],
+        ])->first();
+
+        // If there is no campaign with that title, create one.
+        if (! $campaign) {
+            $campaign = new Campaign;
+
+            $campaign->internal_title = $request['internal_title'];
+            $campaign->start_date = $request['start_date'];
+            $campaign->end_date = $request['end_date'];
+            $campaign->save();
+        }
+
+        // @TODO: return redirect()->route('campaigns.show', $campaign->id);
```

Comment:
I don't think it is necessary but just want to double check - do we want to log anything when a campaign is created, updated or deleted?

Reply:
@chloealee hmm not if we don't do that for posts and signups I don't think?

Reply:
@katiecrane we do for [signups](https://github.com/DoSomething/rogue/blob/master/app/Managers/SignupManager.php#L53) and [posts](https://github.com/DoSomething/rogue/blob/master/app/Managers/PostManager.php#L65) but I feel like that is more so to make sure everything is working ok through the ecosystem (from Phoenix->Rogue etc). 

For campaigns, I'm not sure this is necessary to log since we can see in the database the timestamps if a campaign is created?

Reply:
@chloealee eh, my thought is that there are way less campaigns than signups so couldn't hurt to log this information